src/arc-drop.md

```diff
@@ -0,0 +1,101 @@
+# Dropping
+
+We now need a way to decrease the reference count and drop the data once it is
+low enough, otherwise the data will live forever on the heap.
+
+To do this, we can implement `Drop`.
+
+Basically, we need to:
+1. Get the `ArcInner` value of the `Arc`
+2. Decrement the reference count
+3. If there is only one reference remaining to the data, then:
+4. Atomically fence the data to prevent reordering of the use and deletion of
+   the data, then:
+5. Drop the inner data 
+
+We'll need to import the trait:
```

Comment:
The `Drop` trait is in [the standard library's prelude](https://doc.rust-lang.org/std/prelude/#prelude-contents), so it doesn't need explicitly imported, you can just write `impl Drop for Foo { ... }` anywhere.

Reply:
ah ok